### PR TITLE
Truecolor smooth line fix for #721

### PR DESCRIPTION
**Pull request by zx64**
(Also has #720)

This addresses the basic issue of truncating data from the truecolor colormap and then incorrectly using that to index into the colormap a second time.
Duplicating the definition of PUTDOT isn't ideal, so if you'd prefer me to shuffle that around let me know.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -32,6 +32,8 @@ find_package(SDL2 2.0.7)
 find_package(SDL2_mixer 2.0.2)
 find_package(SDL2_net 2.0.0)
 
+option(CRISPY_TRUECOLOR "True color rendering" OFF)
```
Comment (fabiangreffrath) [inline CMakeLists.txt:35]: Ew, now this also contains the CMake fix that I just merged.

File: src/doom/am_map.c [hunk 2]
```diff
@@ -1278,6 +1278,11 @@ static void AM_drawFline_Smooth(fline_t* fl, int color)
         X0 = X1;
         X1 = Temp;
     }
+#ifdef CRISPY_TRUECOLOR
+    /* We have already applied the colormap lookup in AM_LevelInit so can directly copy the values */
+#undef PUTDOT
```
Comment (fabiangreffrath) [inline src/doom/am_map.c:1283]: The `#undef` is a bit too ugly. Could we get around without it, i.e. with a single definition of `PUTDOT`?

**Review by fabiangreffrath — CHANGES_REQUESTED** [review]

**Comment by zx64** [comment]
Ok, I've created a `PUTDOT_RAW` (name subject to bikeshedding) that has the underlying operation that the TC/not-TC `PUTDOT` definitions forward onto.

**Comment by fabiangreffrath** [comment]
Thank you!

**Pull request merged**